src/warp.jl

```diff
@@ -100,3 +100,46 @@ function warp(img::AbstractArray, tform, args...)
     etp = box_extrapolation(img, args...)
     warp(etp, tform)
 end
+
+"""
+    imrotate(img, angle; fill = NaN, method = "nearest", bbox = "loose") -> imgout
+
+Rotate image `img` by `angle` degrees in a counterclockwise direction around its center point. To rotate the image clockwise, specify a negative value for angle.
+
+# Arguments
+- `img::AbstractArray`: image to be rotated
+- `angle::Real`: amount of rotation in degrees. The rotation is in counterclockwise direction.
+- `fill::Real`: filling value for pixel outside of original image. Defaults: `NaN` if the element type supports it, and `0` otherwise
+- `method::String = "nearest"`: the interpolation method used in rotation. Options are "nearest" or "bilinear"
+- `bbox::String = "loose"`: the bounding box defines the size of output image. If `bbox = "crop"`, then output image `imgout` are cropped to has the same size of input image `img`. If `bbox = "loose"`, then `imgout` is large enough to contain the entire rotated image; `imgout` is larger than `img`.
+
+See also [`warp`](@ref).
+"""
+function imrotate(img::AbstractArray{T}, angle::Real; fill = NaN, method::String = "nearest", bbox::String = "loose")::AbstractArray{T} where T
+
+    # FIXME: Bicubic method does not support N0f8 type.
+    args = []
+    push!(args,
+        if method == "nearest"
+            Constant()
+        elseif method == "bilinear"
+            Linear()
```

Comment:
I am skeptical of this. Why do we need this string glue code when we have the types. In general I think we should avoid string option parameter unless we really need them. I know @timholy has chosen to strings for the padding in `imfilter`. Maybe he has a different opinion on this?

Reply:
In my opinion, the reason is simple.

It is kind of inconvenient writing a two line `warp` code for a simple `imrotate` job, not to mention one has to explicitly `using CoordinateTransformations, Interpolations` for this work.

Using String parameters is my personal old-habit from MATLAB, if you have a better idea of implementing this without String keywords, I'm very happy to change it; actually, I find it not concise in parsing these parameters.

Reply:
See the pattern used for `degree` in `src/interpolations.jl` (in this package) for a good alternative here.
Don't forget to add `imrotate` to the exports list in `src/Interpolations.jl` and drop the reference to a deleted file in `test/runtests.jl`.